File: src/CalendarPrinter/Logic/GenerationFactory.cs
using CalendarPrinter.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CalendarPrinter.Logic
{
    class GenerationFactory
    {
        public ICalendarGenerator Create(CalendarOutputStyle style, CalendarOutputFormat format)
        {
            switch (format)
            {
                case CalendarOutputFormat.HTML:
                    return CreateHTMLStyleGenerator(style);
                case CalendarOutputFormat.PNG:
                    return CreatePNGStyleGenerator(style);
                case CalendarOutputFormat.SVG:
                    return CreateSVGStyleGenerator(style);
                default:
                    throw new ArgumentOutOfRangeException(nameof(format));
            }
        }

        private ICalendarGenerator CreatePNGStyleGenerator(CalendarOutputStyle style)
        {
            throw new ArgumentOutOfRangeException(nameof(style));
        }

        private ICalendarGenerator CreateSVGStyleGenerator(CalendarOutputStyle style)
        {
            switch (style)
            {
                case CalendarOutputStyle.MonthlyLandscape:
                    return new MonthlyLandscapeSVGGenerator();
                default:
                    throw new ArgumentOutOfRangeException(nameof(style));
            }
        }

        private ICalendarGenerator CreateHTMLStyleGenerator(CalendarOutputStyle style)
        {
            switch (style)
            {
                case CalendarOutputStyle.MonthlyLandscape:
                    return new MonthlyLandscapeHTMLGenerator();
                case CalendarOutputStyle.MonthlyPortrait:
                    return new MonthlyPortraitHTMLGenerator();
                default:
                    throw new ArgumentOutOfRangeException(nameof(style));
            }
        }
    }
}

File: src/CalendarPrinter/GenerationFactory.py
from .CalendarOutputFormat import CalendarOutputFormat
from .CalendarOutputStyle import CalendarOutputStyle
from .CalendarGenerator import CalendarGenerator
from .MonthlyLandscapeSVGGenerator import MonthlyLandscapeSVGGenerator
from .MonthlyLandscapeHTMLGenerator import MonthlyLandscapeHTMLGenerator
from .MonthlyPortraitHTMLGenerator import MonthlyPortraitHTMLGenerator

class GenerationFactory:
    def __init__(self):
        pass
    
    def Create(self, style: CalendarOutputStyle, format: CalendarOutputFormat) -> CalendarGenerator:
        match format:
            case CalendarOutputFormat.HTML:
                return self._CreateHTMLStyleGenerator(style)
            case CalendarOutputFormat.SVG:
                return self._CreateSVGStyleGenerator(style)
        
        raise NotImplementedError()
    
    def _CreateHTMLStyleGenerator(self, style: CalendarOutputStyle) -> CalendarGenerator:
        match style:
            case CalendarOutputStyle.MonthlyLandscape:
                return MonthlyLandscapeHTMLGenerator()
            case CalendarOutputStyle.MonthlyPortrait:
                return MonthlyPortraitHTMLGenerator()
        
        raise NotImplementedError()
        
    def _CreateSVGStyleGenerator(self, style: CalendarOutputStyle) -> CalendarGenerator:
        match style:
            case CalendarOutputStyle.MonthlyLandscape:
                return MonthlyLandscapeSVGGenerator()
        
        raise NotImplementedError()